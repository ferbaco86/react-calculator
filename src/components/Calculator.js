import React, { useState } from 'react';
import styled from 'styled-components';
import ButtonPanel from './ButtonPanel';
import Display from './Display';
import NavBar from './NavBar';
import calculate from '../logic/calculate';

const CalculatorContainer = styled.div`
display:flex
flex-direction:column;
width: 700px;
border: 4px solid white`;

const ContentContainer = styled.section`
display: flex;
justify-content: space-between;
padding: 4.5rem`;

const CalculatorTitle = styled.h1`
font-size: 3rem;
color: white;
`;

const Calculator = () => {
  const [state, setState] = useState({ total: null, next: null, operation: null });
  // eslint-disable-next-line no-unused-vars
  const { total, next, operation } = state;

  const handleClick = buttonName => {
    setState(calculate(state, buttonName));
  };

  return (
    <>
      <NavBar />
      <ContentContainer>
        <CalculatorTitle>Let&#39;s do some math!</CalculatorTitle>
        <CalculatorContainer>
          <Display result={next || total || '0'} />
          <ButtonPanel clickHandler={handleClick} />
        </CalculatorContainer>
      </ContentContainer>
    </>
  );
};

export default Calculator;
