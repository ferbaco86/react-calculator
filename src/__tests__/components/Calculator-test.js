import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import {
  render, fireEvent, screen,
} from '@testing-library/react';
import Calculator from '../../components/Calculator';

it('renders correctly', () => {
  const tree = renderer.create(<BrowserRouter><Calculator /></BrowserRouter>).toJSON();
  expect(tree).toMatchSnapshot();
});

describe('Calculator Components', () => {
  test('Have a Title', () => {
    render(<BrowserRouter><Calculator /></BrowserRouter>);

    expect(screen.getAllByRole('heading')[1]).toHaveTextContent("Let's do some math!");
  });

  test('Show a number on the Display', () => {
    render(<BrowserRouter><Calculator /></BrowserRouter>);
    fireEvent.click(screen.getByText('9'));
    expect(screen.getAllByRole('heading')[2]).toHaveTextContent('9');
  });

  test('Using + display an addition', () => {
    render(<BrowserRouter><Calculator /></BrowserRouter>);
    fireEvent.click(screen.getByText('3'));
    fireEvent.click(screen.getByText('+'));
    fireEvent.click(screen.getByText('4'));
    fireEvent.click(screen.getByText('='));
    expect(screen.getAllByRole('heading')[2]).toHaveTextContent('7');
  });
  test('Using - display a substraction', () => {
    render(<BrowserRouter><Calculator /></BrowserRouter>);
    fireEvent.click(screen.getByText('3'));
    fireEvent.click(screen.getByText('-'));
    fireEvent.click(screen.getByText('4'));
    fireEvent.click(screen.getByText('='));
    expect(screen.getAllByRole('heading')[2]).toHaveTextContent('-1');
  });
  test('Using x display a multiplication', () => {
    render(<BrowserRouter><Calculator /></BrowserRouter>);
    fireEvent.click(screen.getByText('3'));
    fireEvent.click(screen.getByText('x'));
    fireEvent.click(screen.getByText('4'));
    fireEvent.click(screen.getByText('='));
    expect(screen.getAllByRole('heading')[2]).toHaveTextContent('12');
  });
  test('Using รท display a division', () => {
    render(<BrowserRouter><Calculator /></BrowserRouter>);
    fireEvent.click(screen.getByText('4'));
    fireEvent.click(screen.getByText('รท'));
    fireEvent.click(screen.getByText('2'));
    fireEvent.click(screen.getByText('='));
    expect(screen.getAllByRole('heading')[2]).toHaveTextContent('2');
  });
  test('Using % display a percentage', () => {
    render(<BrowserRouter><Calculator /></BrowserRouter>);
    fireEvent.click(screen.getByText('4'));
    fireEvent.click(screen.getByText('%'));
    fireEvent.click(screen.getByText('='));
    expect(screen.getAllByRole('heading')[2]).toHaveTextContent('0.04');
  });
  test('Using AC returns to 0', () => {
    render(<BrowserRouter><Calculator /></BrowserRouter>);
    fireEvent.click(screen.getByText('4'));
    fireEvent.click(screen.getByText('AC'));
    expect(screen.getAllByRole('heading')[2]).toHaveTextContent('0');
  });
});
