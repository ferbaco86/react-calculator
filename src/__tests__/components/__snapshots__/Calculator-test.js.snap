// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <nav
    className="sc-eCssSg Gekms"
  >
    <h1
      className="sc-jSgupP iDfkkF"
    >
      Math Magicians
    </h1>
    <div
      className="sc-gKsewC byXJbQ"
    >
      <a
        className="sc-iBPRYJ gGhErT"
        href="/"
        onClick={[Function]}
      >
        Home 
      </a>
      <a
        className="sc-iBPRYJ gGhErT"
        href="/calculator"
        onClick={[Function]}
      >
        Calculator 
      </a>
      <a
        className="sc-iBPRYJ gGhErT"
        href="/quotes"
        onClick={[Function]}
      >
        Quotes 
      </a>
    </div>
  </nav>,
  <section
    className="sc-pFZIQ cVGtXw"
  >
    <h1
      className="sc-jrAGrp fppAqa"
    >
      Let's do some math!
    </h1>
    <div
      className="sc-fubCfw gPJKOk"
    >
      <div
        className="sc-dlfnbm eSRZFs"
      >
        <h1
          className="sc-hKgILt gKwUdW"
        >
          0
        </h1>
      </div>
      <div
        className="sc-gsTCUz hUkXMX"
      >
        <button
          className="sc-bdfBwQ iodbGc"
          onClick={[Function]}
          type="button"
        >
          AC
        </button>
        <button
          className="sc-bdfBwQ iodbGc"
          onClick={[Function]}
          type="button"
        >
          +/-
        </button>
        <button
          className="sc-bdfBwQ iodbGc"
          onClick={[Function]}
          type="button"
        >
          %
        </button>
        <button
          className="sc-bdfBwQ gsJDyK"
          onClick={[Function]}
          type="button"
        >
          รท
        </button>
      </div>
      <div
        className="sc-gsTCUz hUkXMX"
      >
        <button
          className="sc-bdfBwQ iodbGc"
          onClick={[Function]}
          type="button"
        >
          7
        </button>
        <button
          className="sc-bdfBwQ iodbGc"
          onClick={[Function]}
          type="button"
        >
          8
        </button>
        <button
          className="sc-bdfBwQ iodbGc"
          onClick={[Function]}
          type="button"
        >
          9
        </button>
        <button
          className="sc-bdfBwQ gsJDyK"
          onClick={[Function]}
          type="button"
        >
          x
        </button>
      </div>
      <div
        className="sc-gsTCUz hUkXMX"
      >
        <button
          className="sc-bdfBwQ iodbGc"
          onClick={[Function]}
          type="button"
        >
          4
        </button>
        <button
          className="sc-bdfBwQ iodbGc"
          onClick={[Function]}
          type="button"
        >
          5
        </button>
        <button
          className="sc-bdfBwQ iodbGc"
          onClick={[Function]}
          type="button"
        >
          6
        </button>
        <button
          className="sc-bdfBwQ gsJDyK"
          onClick={[Function]}
          type="button"
        >
          -
        </button>
      </div>
      <div
        className="sc-gsTCUz hUkXMX"
      >
        <button
          className="sc-bdfBwQ iodbGc"
          onClick={[Function]}
          type="button"
        >
          1
        </button>
        <button
          className="sc-bdfBwQ iodbGc"
          onClick={[Function]}
          type="button"
        >
          2
        </button>
        <button
          className="sc-bdfBwQ iodbGc"
          onClick={[Function]}
          type="button"
        >
          3
        </button>
        <button
          className="sc-bdfBwQ gsJDyK"
          onClick={[Function]}
          type="button"
        >
          +
        </button>
      </div>
      <div
        className="sc-gsTCUz hUkXMX"
      >
        <button
          className="sc-bdfBwQ iodbKq"
          onClick={[Function]}
          type="button"
        >
          0
        </button>
        <button
          className="sc-bdfBwQ iodbGc"
          onClick={[Function]}
          type="button"
        >
          .
        </button>
        <button
          className="sc-bdfBwQ gsJDyK"
          onClick={[Function]}
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </section>,
]
`;
